src/shared/utils.ts

```diff
@@ -306,3 +307,24 @@ export function bindLetterKeymap(
         [prefix + letter.toUpperCase()]: command,
     };
 }
+
+/**
+ * Tests whether a string is a valid tag name.
+ * @param tagName The string to test
+ * @param allowNonAscii Whether to allow non-ascii characters in the tag name
+ */
+export function validateTagName(
```

Comment:
General tag name validation option was recently added in 6ae693694c08be9afbad7f0668e0043ec747cef2 for a completely unrelated reason, but works great for this use case. Let's drop this method and use that one instead.